pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -277,8 +277,6 @@ func (p *PriorityQueue) isPodBackingOff(pod *v1.Pod) bool {
 // backoffPod checks if pod is currently undergoing backoff. If it is not it updates the backoff
 // timeout otherwise it does nothing.
 func (p *PriorityQueue) backoffPod(pod *v1.Pod) {
-	p.podBackoff.CleanupPodsCompletesBackingoff()
```

Comment:
Removing this may break PR #74157, hence issue #73176 can surface again.

As what you explained in the PR description, we can propose a `p.CleanupPodsCompletesBackingoff()` function to ensure (1) p.pbm.clearPodBackoff(pod) (2) p.clearPodBackoff(pod) are enforced for every expired pod.

WDYT?

cc/ @misterikkit

Reply:
Yes, that line is important. Otherwise a pending pod in backoffQ won't get the chance to reset its expiry, and hence would be retried every 10s eventually.

However, removing this line means we need to manually reset expiry in every occurrence of moving pod from backoffQ/unschedulableQ to activeQ; In addition to `flushBackoffQCompleted`, here are some examples:

- MoveAllToActiveQueue()
- movePodsToActiveQueue(podInfoList []*framework.PodInfo)
- ...

I'm not 100% sure it's worth the complexity.